Modules/_xxsubinterpretersmodule.c

```diff
@@ -2366,6 +2531,50 @@ PyDoc_STRVAR(channel_recv_doc,
 \n\
 Return a new object from the data at the from of the channel's queue.");
 
+static PyObject *
+channel_send_wait(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"cid", "obj", "timeout", NULL};
+    int64_t cid;
+    PyObject *obj;
+    int64_t timeout = -1;
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O&O|l:channel_send_wait",
+                                     kwlist, channel_id_converter,
+                                     &cid, &obj, &timeout)) {
+        return NULL;
+    }
+
+    void *wait_lock = _channelitem_wait_lock_new();
+    if (_channel_send(&_globals.channels, cid, obj, wait_lock) != 0) {
+        return NULL;
+    }
```

Comment:
If `timeout` is 0 then isn't this the same behavior as `channel_send()`?  In that case you can add a short-circuit:

```suggestion
    if (timeout == 0) {
        if (_channel_send(&_globals.channels, cid, obj, NULL) != 0) {
            return NULL;
        }
        Py_RETURN_NONE;
    }

    void *wait_lock = _channelitem_wait_lock_new();
    if (_channel_send(&_globals.channels, cid, obj, wait_lock) != 0) {
        return NULL;
    }
```